docs/userstylesyml.md

```diff
@@ -0,0 +1,154 @@
+     color: <color>
+     icon: <icon>                                    # OPTIONAL
+     readme:
+       app-link: "<url>"
+       usage: |+                                     # OPTIONAL
+         > [!NOTE]
+         > <your notes>
+       faq:                                          # OPTIONAL
+         - question: <question>
+           answer: <answer>
+       current-maintainers: [*<github-username>]
+       past-maintainers: [*<github-username>]        # OPTIONAL
+   ```
+
+### Adding yourself as a maintainer
+
+To add yourself as a maintainer, you need to add your GitHub username to the
+`collaborators` array in the `userstyles.yml` file. This file is located in the
+`scripts` directory. The `collaborators` array, is an array of objects, each
+object should contain a `name` and a `url` key. And optionally a `name` key,
+which is your preferred name.
+
+```yaml
+collaborators:
+  - &your-username
+    name: Your Preferred Name
+    url: https://github.com/your-username
+```
+
+Then you need to add your username to the `current-maintainers` array in for the
+userstyle you want to maintain, in this case `youtube`..
+
+```yaml
+userstyles:
+  youtube:
+    name: YouTube
+    categories: [entertainment, social_networking, photo_and_video]
+    icon: youtube
+    color: red
+    readme:
+      app-link: "https://youtube.com"
+      current-maintainers: [*isabelroses, *your-username]
+      past-maintainers: [*elkrien]
+```
+
+If the change is accepted, you will receive a invite to the `catppuccin` organization, 
+and become a member of the [`userstyles-maintainers` team](https://github.com/orgs/catppuccin/teams/userstyles-maintainers).
+As a maintainer you will be expected to:
+  - Review and merge PRs for the userstyle you maintain.
+  - Keep the userstyle up to date with the latest changes, if you no longer wish to maintain the userstyle see [Removing yourself as a maintainer](#removing-yourself-as-a-maintainer).
+  - Follow our code of conduct and guidelines.
+
+### Removing yourself as a maintainer
+
+To remove yourself as a maintainer, you need to remove your username from the
+`current-maintainers` array for the userstyles you wish to no longer maintain,
+and add yourself to the `past-maintainers` array. In the example of youtube
+below, `elkrien` has been removed from the `current-maintainers` array and added
+to the `past-maintainers` array.
+
+```yaml
+userstyles:
+  youtube:
+    name: YouTube
+    categories: [entertainment, social_networking, photo_and_video]
+    icon: youtube
+    color: red
+    readme:
+      app-link: "https://youtube.com"
+      current-maintainers: [*isabelroses]
+      past-maintainers: [*elkrien]
+```
+
+If you prefer not to be listed as a past maintainer, you can remove yourself for
+maintainers array entirely. Furthermore, removing yourself from the `collaborators` 
+array will also remove you from the `userstyles-maintainers` team.
```

Comment:
This sentence doesn't read very clearly, hard to explain why lol.

Reply:
> This sentence doesn't read very clearly, hard to explain why lol.

maybe it should be the following? (emphasis mine)

"If you prefer not to be listed as a past maintainer, you can remove yourself from
maintainers array entirely. Furthermore, removing yourself from the collaborators 
array will also remove you from the userstyles-maintainers team."